include/ProcessTools.h

```diff
@@ -45,25 +45,93 @@
 
 #include "DataTypes.h"
 
-namespace Mezzanine
-{
-    namespace Testing
+namespace Mezzanine {
+namespace Testing {
+
+SAVE_WARNING_STATE
+SUPPRESS_CLANG_WARNING("-Wpadded")
+
+    /// @brief A simple struct for storing (some) output from a called process.
+    struct MEZZ_LIB CommandResult
     {
-        /// @brief Run a process and capture its console output.
-        /// @param Command The command to attempt to run and capture its output.
-        /// @param TempFileName The file to put all the console output of the command to be transferred back here.
-        /// @return Everything the subprocess emits to the console (stdout or stderr) as a single concatenated string.
-        /// @details This launches a process, redirects its output to a file and reads that file to return that output.
-        /// This is a simple wrapper around std::system, concatenating output redirection to the command. Because of
-        /// this "|", "<" and ">" are not allowed in commands passed to this.
-        Mezzanine::String MEZZ_LIB RunCommand(const Mezzanine::String& Command,
-                                              const Mezzanine::String& TempFileName);
-
-        /// @brief Get all the text in a file.
-        /// @param Filename The file to read.
-        /// @return A string with the contents.
-        Mezzanine::String MEZZ_LIB GetFileContents(const Mezzanine::String& Filename);
-    }// Testing
+        /// @brief The output to cout and cerr from the called process.
+        String ConsoleOutput;
+        /// @brief The code returned when the called process exited.
+        Integer ExitCode = EXIT_FAILURE;
+    };//CommandResult
+
+RESTORE_WARNING_STATE
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Output to String
+
+    /// @brief Launches a different process on the system.
+    /// @note ExecutablePath cannot be empty on Posix systems or this function will fail.
+    /// @param ExecutablePath The path to the executable to be invoked.
+    /// @param Command The command to attempt to run and direct its output.
+    /// @return Returns the ExitCode and Cout/Cerr output of the command that was run.
+    CommandResult MEZZ_LIB RunCommand(const StringView ExecutablePath, const StringView Command);
```

Comment:
Per the test this works:

`Testing::GetCommandOutput("cmake -E echo foo"))`

None of this looks like an "ExecutablePath" would that be more clear as Executable Name?

Reply:
Perhaps This deserves a documentation page, to explain if paths can be passed.

Reply:
Note the second parameter isn't defaulted.  It's using the single Stringview overload of the function which takes one big command.

Reply:
I meant on the name and description ExecutablePath does not make it seem that `cmake` in isolation would be a valid path, and 99+% of the time it won't be. The CMake execuable doesn't exist in the WorkingDirectory so something other than a lookup by path would be at work here. Clearly it is searching the OS PATH and finding the executable elsewhere. 

It seems like this can accept 3 things as the first argument a command that exists where the OS Searches, an absolute path or a relative path. This being called a path does not make that clear in my opinion.

Reply:
I just checked, This is actually already an API Change, I also did a quick grep, we don't use this anywhere else.  I think we can minimize this API.